automation/jleechanorg_pr_automation/automation_utils.py

```
@@ -0,0 +1,312 @@
+    # Default configuration
+    DEFAULT_CONFIG = {
+        "SMTP_SERVER": "smtp.gmail.com",
+        "SMTP_PORT": 587,
+        "LOG_DIR": "~/Library/Logs/worldarchitect-automation",
+        "DATA_DIR": "~/Library/Application Support/worldarchitect-automation",
+        "MAX_SUBPROCESS_TIMEOUT": int(os.getenv("AUTOMATION_SUBPROCESS_TIMEOUT", "300")),  # 5 minutes (configurable)
+        "EMAIL_SUBJECT_PREFIX": "[WorldArchitect Automation]"
+    }
+
+    @classmethod
+    def setup_logging(cls, name: str, log_filename: str = None) -> logging.Logger:
```

Ruff (0.14.5)

PEP 484 prohibits implicit `Optional`

Convert to `Optional[T]`

(RUF013)